### ISSUE TITLE
Fixture applied more than once - when running multiple tests

### ISSUE THREAD

**Issue by stephenlagree** [smarie/python-pytest-cases#126]
It seems I can only use my set of cases with a fixture for one test or else I get the following error

`    "fixture is being applied more than once to the same function"
E   ValueError: fixture is being applied more than once to the same function`

Here is code that reproduces:

```python
import pytest
from pytest_cases import parametrize_with_cases


@pytest.fixture()
def dependent_fixture():
    return 1


def case_requirement_1(dependent_fixture):
    return 1 + dependent_fixture


def case_requirement_2(dependent_fixture):
    return 2 + dependent_fixture


@parametrize_with_cases("my_requirement",cases=".", prefix="case")
def test_functionality(my_requirement):
    print(my_requirement)


@parametrize_with_cases("my_requirement",cases=".", prefix="case")
def test_functionality_again(my_requirement):
    print(my_requirement)
```

If I remove the second test, it works fine.  Also, if I make the cases not depend on a fixture, it works fine.
It seems like it's including the fixture too many times when the decorator is used more than once.

**Comment by smarie** [comment]
Thanks very much for reporting ! I'll have a look

**Comment by smarie** [comment]
This is related to the way I handle cases requiring fixtures as of today: I create one fixture for each such case, and I register this fixture in the collection host (module, class) of the test function/fixture parametrized with the cases. This was not entirely bullet proof as you discovered: I was not properly creating a copy (hence the error message you receive, saying that the case function that was already turned into a fixture, is once again turned into a fixture).

This requires some major refactoring to handle all 4 situations (case in module/class, test in module/class) but I made progress today. I'll keep you posted.

**Comment by smarie** [comment]
This was a hard one :) It seems to be fixed in 2.2.0 (to be released soon by travis: https://travis-ci.org/github/smarie/python-pytest-cases/builds )
Please feel free to reopen if it is not fixed ! Thanks again for reporting @stephenlagree
